torch/csrc/distributed/rpc/tensorpipe_utils.h

```diff
@@ -43,14 +132,19 @@ struct TensorpipeReadBuffers {
 TORCH_API std::tuple<tensorpipe::Message, TensorpipeWriteBuffers>
 tensorpipeSerialize(
     Message&& rpcMessage,
-    std::vector<c10::DeviceIndex> devices = {});
+    std::vector<c10::DeviceIndex> devices = {},
+    const std::shared_ptr<FullDeviceContext>& =
+        std::make_shared<FullDeviceContext>(/* noCuda */ true));
```

Comment:
Nit: this may be just my opinion, but I think that passing a const ref to a (smart) pointer isn't idiomatic. It would be if the function needs to consume the pointer itself (e.g., by creating a weak_ptr out of that shared_ptr). But here the function only consumes the value pointed to by the pointer, hence the function doesn't need to know how that value is managed and kept alive, it just needs a reference to the value itself, not to its container.

See https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/ for a writing that says the above but in a better way.
```suggestion
    const FullDeviceContext&);
```

Reply:
will address this one in follow up PRs.